src/assets/Generator.Shared/OperationOrResponseInternals.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Linq;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// This implements the ARM scenarios for LROs. It is highly recommended to read the ARM spec prior to modifying this code:
+    /// https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#asynchronous-operations
+    /// Other reference documents include:
+    /// https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-long-running-operation
+    /// https://github.com/Azure/adx-documentation-pr/blob/master/sdks/LRO/LRO_AzureSDK.md
+    /// </summary>
+    /// <typeparam name="T">The final result of the LRO.</typeparam>
+    internal class OperationOrResponseInternals<T>
```

Comment:
I thought we've decided to inherit from OperationInternals and override members?

Reply:
the challenge comes in that the constructor assumes a bunch of stuff is coming in that isn't in the response case.  So we would have to make a bunch of stuff nullable even when we really don't intend it to be nullable in the case where OperationInternals is actually used